### PR TITLE
`auto_date_histogram` aggregation

### PR DESCRIPTION
**Pull request by trzysiek**
Only `bucket_nr=1` case is handled, `key` value for `auto_date_histogram`'s only bucket is extracted from `where` clause via a simple visitor.

### CODE REVIEW

File: quesma/model/where_visitor.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package model
+
+// TODO: it's not 100% full/proper implementation, but works in the client case
+func FindTimestampLowerBound(expr Expr) (InfixExpr, bool) {
```
[the remark this answers is not in the document]
Comment (trzysiek) [reply quesma/model/where_visitor.go:4]: Maybe add that as TODO to do in next PR, when we merge all of those?
We only handle `bucket_nr=1` case and there we don't need the upper bound.

[the remark this answers is not in the document]
Comment (jakozaur) [reply quesma/model/where_visitor.go:4]: Ok

File: quesma/queryparser/pancake_transformer.go [hunk 2]
```diff
@@ -374,6 +374,52 @@ func (a *pancakeTransformer) createTopHitAndTopMetricsPancakes(pancake *pancakeM
 	return
 }
 
+func (a *pancakeTransformer) transformAutoDateHistogram(layers []*pancakeModelLayer, whereClause model.Expr) {
+	for _, layer := range layers {
+		if layer.nextBucketAggregation != nil {
+			if autoDateHistogram, ok := layer.nextBucketAggregation.queryType.(*bucket_aggregations.AutoDateHistogram); ok {
+				lowerBoundsInWhere := model.FindLowerBounds(whereClause)
```
Comment (jakozaur) [inline quesma/queryparser/pancake_transformer.go:381]: Expand this method, but cut some code from here.

File: quesma/queryparser/pancake_transformer.go [hunk 3]
```diff
@@ -374,6 +374,52 @@ func (a *pancakeTransformer) createTopHitAndTopMetricsPancakes(pancake *pancakeM
 	return
 }
```
Comment (jakozaur) [inline quesma/queryparser/pancake_transformer.go:376]:
```suggestion
Auto date histogram is a date histogram, that automatically creates buckets based on time range. To do that we need parse where which happens in this method.
```

**Pull request merged**